### ISSUE TITLE
Possibility to access all ngbTabs and ngbSlides

### ISSUE THREAD

**Issue by kubica** [ng-bootstrap/ng-bootstrap#2514]
I have build a combination of ngbTabset and ngbCarousel. When the current tab is changed i also want to change the current slide and similar in the other direction (both have the same number of items). This is pretty simple if we only have one tabset on our side and only one carousel because the ids are set incremental.

**BUT** this is very hard when we have multiple ngbTabsets on our page because we need to be able to check the position of our current element in the overall ngbTabset / ngbCarousel and select this current position in the corresponding ngbCarousel / ngbTabset.

### Bug description:

The current event triggered contains:

<img width="459" alt="screen shot 2018-07-11 at 11 49 50" src="https://user-images.githubusercontent.com/4146400/42564346-cd421a0c-8500-11e8-8fec-46614c6e70f4.png">

and the current tabset contains:

<img width="960" alt="screen shot 2018-07-11 at 11 50 02" src="https://user-images.githubusercontent.com/4146400/42564354-d2ea4024-8500-11e8-8d16-7c445425194b.png">

In order to check the current position i either need it to be given or manually find it (which was my approach).

The only way you can access all tabs is using `tabset.tabs._results` or am i wrong? This is possible in dev environments but if i build my app using AoT compilation and rollup this ends in

> error TS2341: Property '_results' is private and only accessible within class 'QueryList<T>'.

Is there any way to access all tabs and slides **OR** got the position of the current element **OR** do i manually use an index?

**Comment by pkozlowski-opensource** [comment]
I'm sorry but it is not clear from the description provided what is your exact use-case / problem. Try to reproduce it in a plunker / stackblitz (you can fork one from our demo page to get started).

**Comment by pkozlowski-opensource** [comment]
`tabset.tabs._results` is a private data structure and should _not_ be accessed by widget users.

**Comment by kubica** [comment]
It's very hard to reproduce and i could build a solution using some hacks to got the current position but i just wondered if **(1)** there is a way to either get the full list of all tabs (inside ngbTabSet) or **(2)** to get the current element position inside the array (in the tabChangeEvent) - not just the name of the tab?!

The problem is pretty simple. I have multiple ngbTabSets and need to get the position of the current tab when it's changed. Due to the fact that the ngbTabSets have variable sizes i can not use incremental number to map it to the current position of the tabs. Is there a way to manage this?

**Comment by benouat** [comment]
Thank you for the explanation, but I must admit that even with it, I still don't get what you want to do 🙁

**Comment by kubica** [comment]
@pkozlowski-opensource , @benouat Thanks for the comments. I will try it again, this time using a stackblitz: [https://stackblitz.com/edit/angular-1rmyfu](https://stackblitz.com/edit/angular-1rmyfu). Hope this will help.

What is want to accomplish is that i have 2 tabsets (or 1 tabset and 1 carousel) with the same length of tabs (or slides) and the should be sync with each other. When pressing 3 in the stackblitz the other tabset should also select tab 3 (or slide 3).

I know that this is not the correct usage of tabsets but is it still possible to do? Is there any way to select the current tab (slide) of a tabset (slideset) using a position instead of ids?

**Comment by pkozlowski-opensource** [comment]
@kubica I don't think it is possible to do what you are trying to do currently. To sync 2 tabsets we would have to have a different event that is fired _after_ selection is done, otherwise we are getting into an infinite loop.

The mentioned "after selection" event was requested in another issue (#2622) so I'm going to close this one as a duplicate.

**Comment by kubica** [comment]
For someones interest, I solved it by the following workaround:
- every tabset has something like this: `#t1="ngbTabset" (tabChange)="changeTab($event, t1, t2)`
- it is possible to request the id of the first element of the tabsets (`tabset.tabs.first.id` where tabset is t1 or t2)
- I parse the "number" of these id's (for example: Tabset 1 - "ngb-tab-5" -> `tab1 = 5`, Tabset 2 - "ngb-tab-10" -> `tab2 = 10`) and calc the difference of the currentTabset to the event's nextId (`event.nextId` - for example: "ngb-tab-8" -> 8 = difference of +3).
- in the next step i set the id of the second tabset (`tabset2.select("ngb-tab-"+(tab3+diff)`) and set in parallel a variable to do not end in an infinite loop ("pending = true")
- this variable is used in the top of the function and resetted on the next changeTab call which is done on `tabset2.select`
- the tabsets are synced now
